### DISCUSSION TITLE
Citation of R packages directly on Quarto

### DISCUSSION THREAD

**Discussion by jospueyo** [quarto-dev/quarto-cli#7971] [Feature Requests]
### Description

How nice it would be having a function to directly cite R packages in Quarto? Something like:

`r citation(<package>)` and the output becoming a reference as any other reference. 

I tried doing a custom function with the following workflow: 

1. Extract the information using `citation()`
2. Use `rmarkdown::metadata$bibliography` to obtain the bibliography file.
3. Use `writeLines()` to add the reference to the bib file.
4. Output a markdown reference @<package>.

However, the problem is that the changes in bibliography are persistent so it does not work when render the file a second time. I would be keen to do a pull request if I receive some guidance on how to surpass this problem.

**Comment by mcanouil** [comment]
See:
- #5806

**Comment by arnaudgallou** [comment]
Here're existing solutions to do this:

- [pakret](https://github.com/arnaudgallou/pakret)
- [grateful](https://github.com/Pakillo/grateful)
- [softbib](https://github.com/vincentarelbundock/softbib)

Being the developer of pakret, I'll naturally recommend it over the other two. Now to be a bit more objective, pakret is simpler to use and a lot more flexible. Compared to grateful and softbib, pakret offers the following advantages:
- allows inline citations
- uses a template system, giving full control on how to cite packages in the text (you could use it in a text written in Chinese for example) 
- doesn't overwrite bib files so you can use a single file to reference both papers and packages
- can write references in different bib files 
- doesn't require any parametrization to be used
- uses a single reference by package to avoid bloating the reference list
- uses consistent package keys to reference R packages
